Add example for using the glasbey colormap with napari

jni opened the pull request:
This adds a gallery example showing how one can use the glasbey colormap with napari.

<img width="892" alt="Screenshot 2024-12-26 at 6 15 09 pm" src="https://github.com/user-attachments/assets/92a20213-af03-4fd2-bb4f-d35de1d96cb7" />

jni committed:
Add example for using the glasbey colormap with napari

psobolewskiPhD commented on examples/glasbey-colormap.py:
```diff
@@ -0,0 +1,81 @@
+layers using a well-known discrete colormap such as Glasbey [2]_.
+
+In this example, we demonstrate displaying segmentations using custom colormaps
+with the help of the glasbey Python library [3]_. We display a segmentation
```
Is it worth explicitly saying the user will need to install it?

jni replied:
imho a gallery example is the wrong place, but I could do if people feel strongly

willingc committed:
Merge branch 'main' into glasbey

jni committed:
Add glasbey to gallery dependencies

jni committed:
Fix plot_2d_edge_meshes when numba installed

This numba error was being masked by the dispatch machinery
when numba was not installed — which happened to be the case
when running the gallery tests.

jni committed:
Install docs and gallery deps in circleCI

jni committed:
Make docs-build rather than docs

jni committed:
Add missing dependencies from napari/docs#114

psobolewskiPhD committed:
Merge branch 'main' into glasbey

willingc commented on examples/glasbey-colormap.py:
```diff
@@ -0,0 +1,81 @@
+.. [1] https://github.com/napari/napari/issues/454
+.. [2] Colour displays for categorical images. Chris Glasbey, Gerie van der
+       Heijden, Vivian F. K. Toh, and Alision Gray. (2007)
+       :DOI:`10.1002/col.20327`
```
I think if we remove the `:DOI:` role that this will pass. Not sure if this is a role that we support.

jni replied:
Oh! We've had this in skimage for so long that I thought it was built into sphinx! 😂 

This is what we would need to use it:

https://github.com/scikit-image/scikit-image/pull/3367/files#diff-f31881d2140a5aa15bb25932e98afa77b57930e1c589ca9785c82c1f92a718ae

Since it's not even a properly packaged extension, I am going to go ahead and remove it. Thanks for spotting it @willingc!

jni replied:
```suggestion
       DOI:10.1002/col.20327
```

jni committed:
Remove usage of non-standard DOI role

psobolewskiPhD commented on examples/glasbey-colormap.py:
```diff
@@ -0,0 +1,81 @@
+layers using a well-known discrete colormap such as Glasbey [2]_.
+
+In this example, we demonstrate displaying segmentations using custom colormaps
+with the help of the glasbey Python library [3]_. We display a segmentation
```
```suggestion
with the help of the glasbey Python library [3]_ that you can install using conda or pip. We display a segmentation
```

psobolewskiPhD replied:
I feel like explicitly mentioning that one needs to install this is important, because otherwise to me it suggests it should just work with napari out of the box and yet it doesn't.

willingc replied:
Maybe leave off conda or pip and end your suggestion after install. I'm pretty much using mostly uv and pixi these days.

psobolewskiPhD replied:
I was going off of our other examples, e.g.:
https://github.com/napari/napari/blob/68de9b0fad0cc5e7198dfa19565d334245d8e345/examples/surface_timeseries_.py#L31-L33

willingc replied:
Fair enough for this PR. We probably will need to update all within the next year :wink:

psobolewskiPhD commented on examples/glasbey-colormap.py:
```diff
@@ -0,0 +1,81 @@
+
+###############################################################################
+# Finally, we display the coins image and the overlaid segmentation. We do this
+# twice to show both colormaps.
```
```suggestion
# in two viewers to show both colormaps.
```

psobolewskiPhD commented on examples/glasbey-colormap.py:
```diff
@@ -0,0 +1,81 @@
+from skimage import data
+from skimage.filters import threshold_otsu
+from skimage.measure import label
+from skimage.morphology import closing, remove_small_objects, square
```
```suggestion
from skimage.morphology import closing, remove_small_objects, footprint_rectangle
```
`square` triggers a deprecation warning

psobolewskiPhD commented on examples/glasbey-colormap.py:
```diff
@@ -0,0 +1,81 @@
+
+# apply threshold
+thresh = threshold_otsu(image)
+bw = closing(image > thresh, square(4))
```
```suggestion
bw = closing(image > thresh, footprint_rectangle((4, 4))
```
`square` triggers deprecation warning

psobolewskiPhD reviewed (changes requested):
I dig this, I know a lot of people coming from Fiji might want to use glasbey.
I do think we should make explicitly clear that `glasbey` needs to be installed for this to work, because the initial paragraph: "A long requested feature" makes it sound like this will work out-of-the–box. (also in other examples like the one with nilearn we make it clear.)
Also, `square` is triggering a deprecation warning for me locally and it also shows in the build docs, so it's probably best to fix that.

jni committed:
Note that the glasbey package needs to be installed

jni committed:
Use numpy to specify square footprint

jni committed:
Specify two viewers in the text

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>
